src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1546 @@
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         *
+         * @implSpec The default implementation is equivalent to
+         * {@link ArbitrarilyJumpableGenerator#jumps(long) jumps}
+         * ({@link Long#MAX_VALUE Long.MAX_VALUE}).
+         */
+        default Stream<ArbitrarilyJumpableGenerator> jumps(double distance) {
+            return Stream.generate(() -> copyAndJump(distance)).sequential();
+        }
+
+        /**
+         * Returns a stream producing the given {@code streamSize} number of new
+         * pseudorandom number generators, each of which implements the
+         * {@link ArbitrarilyJumpableGenerator} interface, produced by jumping
+         * copies of this generator by different integer multiples of the
+         * specified jump distance.
+         *
+         * @param streamSize the number of generators to generate
+         * @param distance   a distance to jump forward within the state cycle
+         *
+         * @return a stream of objects that implement the {@link RandomGenerator} interface
+         *
+         * @throws IllegalArgumentException if {@code streamSize} is less than zero or if
+         *                                  {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         *
+         * @implSpec The default implementation is equivalent to
+         * jumps(distance).limit(streamSize).
+         */
+        default Stream<ArbitrarilyJumpableGenerator> jumps(long streamSize, double distance) {
+            return jumps(distance).limit(streamSize);
+        }
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a very large, fixed distance (typically 2<sup>128</sup> or
+         * more) within its state cycle. The distance used is that returned by
+         * method
+         * {@link ArbitrarilyJumpableGenerator#leapDistance() leapDistance}().
+         */
+        default void leap() { jump(leapDistance()); }
+
+        /**
+         * Copy this generator, jump this generator forward, then return the
+         * copy.
+         *
+         * @param distance a distance to jump forward within the state cycle
+         *
+         * @return a copy of this generator object before the jump occurred
+         *
+         * @throws IllegalArgumentException if {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         *
+         * @implSpec The default implementation copies this, jumps(distance) and then
+         * returns the copy.
+         */
+        default ArbitrarilyJumpableGenerator copyAndJump(double distance) {
```

Comment:
Suggest Override and inheritDoc.

Reply:
jump(double distance) does not occur in superinterface. The ability to jump arbitrarily is specific to ArbitrarilyJumpableGenerator.